src/mgmt/srv_drpc.c

```diff
@@ -747,6 +747,127 @@ ds_mgmt_drpc_pool_get_acl(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	free_ace_list(ace_list, ace_nr);
 }
 
+void
+ds_mgmt_drpc_list_pools(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PoolListReq		*req = NULL;
+	Mgmt__PoolListResp		*resp = NULL;
+	uint8_t				*body;
+	size_t				 len;
+	struct mgmt_list_pools_one	*pools = NULL;
+	size_t				 pools_len = 0;
+	uint64_t			 npools;
+	int			 srbuflen = 32;
+	int				 sri;		/* svcrep buf index */
+	int				 pi;		/* pool index*/
+	int				 ri;		/* svcrep index */
+	int				 rc = 0;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__pool_list_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to unpack req (list pools)\n");
+		mgmt__pool_list_req__free_unpacked(req, NULL);
+		return;
+	}
+
+	D_INFO("Received request to list pools in DAOS system %s\n",
+		req->lp_grp);
+
+	D_ALLOC_PTR(resp);
+	if (resp == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to allocate daos response ref\n");
```

Comment:
Not actually necessary to log here since D_ALLOC_PTR should log everything if it fails